source/type/pt_data.cpp

```diff
@@ -221,6 +222,58 @@ PT_Data::get_stop_point_connection(const StopPoint& from, const StopPoint& to) c
 }
 
 
+nt::JourneyPattern* PT_Data::get_or_create_journey_pattern(const nt::JourneyPatternKey& key){
+    const auto& it = journey_patterns_pool.find(key);
+    if (it != journey_patterns_pool.cend()) {
+        return it->second;
+    }
+    auto new_jp = new JourneyPattern{};
+
+    new_jp->is_frequence = key.is_frequence;
+    new_jp->route = key.route;
+    new_jp->commercial_mode = key.commercial_mode;
+    new_jp->physical_mode = key.physical_mode;
+    new_jp->odt_properties = key.odt_properties;
+    new_jp->name = key.name;
+    new_jp->uri = key.uri + ":adapted-" + std::to_string(journey_patterns.size()) ;
```

Comment:
Will it work if we remove the 2nd sp from a vj a given day and the 3rd sp the day after. Shouldn't we have then 2 different journey pattern with the same uri?